x/gamm/pool-models/stableswap/amm.go

```diff
@@ -180,6 +188,10 @@ func solveCFMMBinarySearch(constantFunction func(sdk.Dec, sdk.Dec) sdk.Dec) func
 	}
 }
 
+func solveCfmm(xReserve, yReserve, yIn sdk.Dec) sdk.Dec {
```

Comment:
Why do we need this wrapper and not use `solveCFMMBinarySearch` directly?

Reply:
Same question here! Maybe we have plans to have different implementations in the future?

Reply:
so we can easily swap the backend implementation, without replacing the function everywhere else

Reply:
I'm not sure I follow, is it because of the time dependencies and other priorities over replacing `solveCfmm` implementation everywhere?

Reply:
@AlpinYukseloglu

Reply:
The idea was to have a general solveCfmm function that abstracts away the specific solving methodology in case we change it in the future. It also felt cleaner to keep it as a general solver function and keep it as the only one instead of having two functions floating around with only one actually used (we would still need to keep the original solver for testing reasons.